src/components/form/form.component.tsx

```diff
@@ -29,11 +57,11 @@ const Form = ({
   height,
   fullWidthButtons = false,
   ...rest
-}) => {
+}: FormProps) => {
   const { isInSidebar } = useContext(SidebarContext);
   const { isInModal } = useContext(ModalContext);
-  const formRef = useRef();
-  const formFooterRef = useRef();
+  const formRef = useRef() as React.MutableRefObject<HTMLFormElement>;
+  const formFooterRef = useRef() as React.MutableRefObject<HTMLDivElement>;
```

Comment:
```suggestion
  const formRef = useRef<HTMLFormElement>(null);
  const formFooterRef = useRef<HTMLDivElement>(null);
```
Is there a reason why these refs need this mutable type? I think ideally we should avoid forcing type assertions where possible, since it could lead us to overlook potential issues.

Reply:
no, there's no good reason, I just needed a stricter type on these refs and this was the first thing I came up with, without thinking too much. Perhaps because I saw this in other components? In any case I agree with you that type assertions aren't ideal and much prefer your suggestion so I will change this :)